spyder/plugins/explorer/utils.py

```diff
@@ -0,0 +1,280 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+Utilities for file explorer widgets.
+"""
+
+# Standard library imports
+from __future__ import print_function
+import glob
+import itertools
+import os
+import sys
+
+# Third party imports
+from qtpy.QtGui import QIcon
+
+# Local imports
+from spyder.config.base import _, get_conf_path
+from spyder.utils import icon_manager as ima
+
+
+def parse_linux_desktop_entry(fpath):
+    """Load data from desktop entry with xdg specification."""
+    from xdg.DesktopEntry import DesktopEntry
+
+    try:
+        entry = DesktopEntry(fpath)
+        entry_data = {}
+        entry_data['name'] = entry.getName()
+        entry_data['icon_path'] = entry.getIcon()
+        entry_data['exec'] = entry.getExec()
+        entry_data['type'] = entry.getType()
+        entry_data['hidden'] = entry.getHidden()
+        entry_data['fpath'] = fpath
+    except Exception:
+        entry_data = {
+            'name': '',
+            'icon_path': '',
+            'hidden': '',
+            'exec': '',
+            'type': '',
+            'fpath': fpath
+        }
+
+    return entry_data
+
+
+def _get_mac_application_icon_path(app_bundle_path):
+    """Parse mac application bundle and return path for *.icns file."""
+    import plistlib
+    contents_path = info_path = os.path.join(app_bundle_path, 'Contents')
+    info_path = os.path.join(contents_path, 'Info.plist')
+
+    pl = {}
+    if os.path.isfile(info_path):
+        try:
+            # readPlist is deprecated but needed for py27 compat
+            pl = plistlib.readPlist(info_path)
+        except Exception:
+            # TODO: Add the specific errors to catch
```

Comment:
I once got an error on Mac but forgot the type of it and then I never got the error again. It was to catch an specific error instead of the generic `Exception` one